Parser/myreadline.c

```diff
@@ -392,9 +392,11 @@ PyOS_Readline(FILE *sys_stdin, FILE *sys_stdout, const char *prompt)
         }
     }
 
-    _PyOS_ReadlineTState = tstate;
-    Py_BEGIN_ALLOW_THREADS
+
     PyThread_acquire_lock(_PyOS_ReadlineLock, 1);
```

Comment:
Would you mind to add a comment explaining the order of the lock and the GIL? Explain that the lock must be acquired first to prevent gh-123321 issue.